<?xml version="1.0" ?>
<sdf version="1.6">
  <world name="obstacles">
    <plugin name="gazebo_ros_factory" filename="libgazebo_ros_factory.so"/>

    <!-- A global light source -->
    <include>
      <uri>model://sun</uri>
    </include>
    <!-- A ground plane -->
    <include>
      <uri>model://ground_plane</uri>
    </include>

    <model name="box_0">
      <static>true</static>
      <link name="link_0">
        <visual name="visual_0">
          <pose>0.1 0.15 0.3 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.3 0.6</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_0">
          <pose>0.1 0.15 0.3 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.3 0.6</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_1">
      <static>true</static>
      <link name="link_1">
        <visual name="visual_1">
          <pose>1.4 0.25 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.8 0.3 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_1">
          <pose>1.4 0.25 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.8 0.3 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_2">
      <static>true</static>
      <link name="link_2">
        <visual name="visual_2">
          <pose>1.9 0.25 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.3 0.3</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_2">
          <pose>1.9 0.25 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.3 0.3</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_3">
      <static>true</static>
      <link name="link_3">
        <visual name="visual_3">
          <pose>4 0.6 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.8 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_3">
          <pose>4 0.6 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.8 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_4">
      <static>true</static>
      <link name="link_4">
        <visual name="visual_4">
          <pose>0.45 0.45 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>0.1 0.1 0.3</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_4">
          <pose>0.45 0.45 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>0.1 0.1 0.3</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_5">
      <static>true</static>
      <link name="link_5">
        <visual name="visual_5">
          <pose>1.25 0.55 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.3 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_5">
          <pose>1.25 0.55 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.3 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_6">
      <static>true</static>
      <link name="link_6">
        <visual name="visual_6">
          <pose>2.8 0.7 0.1 0 0 0</pose>
          <geometry>
            <box>
              <size>0.4 0.2 0.2</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_6">
          <pose>2.8 0.7 0.1 0 0 0</pose>
          <geometry>
            <box>
              <size>0.4 0.2 0.2</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_7">
      <static>true</static>
      <link name="link_7">
        <visual name="visual_7">
          <pose>0.45 0.75 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.1 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_7">
          <pose>0.45 0.75 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.1 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_8">
      <static>true</static>
      <link name="link_8">
        <visual name="visual_8">
          <pose>1.3 0.75 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.6 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_8">
          <pose>1.3 0.75 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.6 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_9">
      <static>true</static>
      <link name="link_9">
        <visual name="visual_9">
          <pose>1.25 0.85 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_9">
          <pose>1.25 0.85 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_10">
      <static>true</static>
      <link name="link_10">
        <visual name="visual_10">
          <pose>2.85 0.85 0.1 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.2</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_10">
          <pose>2.85 0.85 0.1 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.2</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_11">
      <static>true</static>
      <link name="link_11">
        <visual name="visual_11">
          <pose>1.25 0.95 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.7 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_11">
          <pose>1.25 0.95 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.7 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_12">
      <static>true</static>
      <link name="link_12">
        <visual name="visual_12">
          <pose>2.8 1 0.1 0 0 0</pose>
          <geometry>
            <box>
              <size>0.4 0.2 0.2</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_12">
          <pose>2.8 1 0.1 0 0 0</pose>
          <geometry>
            <box>
              <size>0.4 0.2 0.2</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_13">
      <static>true</static>
      <link name="link_13">
        <visual name="visual_13">
          <pose>0.45 1.15 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>0.1 0.3 0.3</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_13">
          <pose>0.45 1.15 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>0.1 0.3 0.3</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_14">
      <static>true</static>
      <link name="link_14">
        <visual name="visual_14">
          <pose>1.2 1.05 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.6 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_14">
          <pose>1.2 1.05 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.6 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_15">
      <static>true</static>
      <link name="link_15">
        <visual name="visual_15">
          <pose>1.25 1.35 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.5 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_15">
          <pose>1.25 1.35 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.5 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_16">
      <static>true</static>
      <link name="link_16">
        <visual name="visual_16">
          <pose>2.8 1.35 0.1 0 0 0</pose>
          <geometry>
            <box>
              <size>0.4 0.3 0.2</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_16">
          <pose>2.8 1.35 0.1 0 0 0</pose>
          <geometry>
            <box>
              <size>0.4 0.3 0.2</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_17">
      <static>true</static>
      <link name="link_17">
        <visual name="visual_17">
          <pose>0.55 1.95 0.25 0 0 0</pose>
          <geometry>
            <box>
              <size>0.1 0.1 0.5</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_17">
          <pose>0.55 1.95 0.25 0 0 0</pose>
          <geometry>
            <box>
              <size>0.1 0.1 0.5</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_18">
      <static>true</static>
      <link name="link_18">
        <visual name="visual_18">
          <pose>2 2.1 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>0.4 0.4 0.3</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_18">
          <pose>2 2.1 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>0.4 0.4 0.3</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_19">
      <static>true</static>
      <link name="link_19">
        <visual name="visual_19">
          <pose>2.65 2.1 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.4 0.3</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_19">
          <pose>2.65 2.1 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.4 0.3</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_20">
      <static>true</static>
      <link name="link_20">
        <visual name="visual_20">
          <pose>0.55 2.05 0.25 0 0 0</pose>
          <geometry>
            <box>
              <size>0.3 0.1 0.5</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_20">
          <pose>0.55 2.05 0.25 0 0 0</pose>
          <geometry>
            <box>
              <size>0.3 0.1 0.5</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_21">
      <static>true</static>
      <link name="link_21">
        <visual name="visual_21">
          <pose>0.55 2.15 0.25 0 0 0</pose>
          <geometry>
            <box>
              <size>0.1 0.1 0.5</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_21">
          <pose>0.55 2.15 0.25 0 0 0</pose>
          <geometry>
            <box>
              <size>0.1 0.1 0.5</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_22">
      <static>true</static>
      <link name="link_22">
        <visual name="visual_22">
          <pose>2.35 2.45 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>1.1 0.3 0.3</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_22">
          <pose>2.35 2.45 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>1.1 0.3 0.3</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_23">
      <static>true</static>
      <link name="link_23">
        <visual name="visual_23">
          <pose>2 2.65 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>0.4 0.1 0.3</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_23">
          <pose>2 2.65 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>0.4 0.1 0.3</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_24">
      <static>true</static>
      <link name="link_24">
        <visual name="visual_24">
          <pose>2.65 2.65 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.3</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_24">
          <pose>2.65 2.65 0.15 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.3</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_25">
      <static>true</static>
      <link name="link_25">
        <visual name="visual_25">
          <pose>0.6 2.75 0.2 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.1 0.4</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_25">
          <pose>0.6 2.75 0.2 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.1 0.4</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_26">
      <static>true</static>
      <link name="link_26">
        <visual name="visual_26">
          <pose>0.6 2.85 0.2 0 0 0</pose>
          <geometry>
            <box>
              <size>0.4 0.1 0.4</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_26">
          <pose>0.6 2.85 0.2 0 0 0</pose>
          <geometry>
            <box>
              <size>0.4 0.1 0.4</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_27">
      <static>true</static>
      <link name="link_27">
        <visual name="visual_27">
          <pose>0.6 2.95 0.2 0 0 0</pose>
          <geometry>
            <box>
              <size>0.6 0.1 0.4</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_27">
          <pose>0.6 2.95 0.2 0 0 0</pose>
          <geometry>
            <box>
              <size>0.6 0.1 0.4</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_28">
      <static>true</static>
      <link name="link_28">
        <visual name="visual_28">
          <pose>0.6 3.05 0.2 0 0 0</pose>
          <geometry>
            <box>
              <size>0.4 0.1 0.4</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_28">
          <pose>0.6 3.05 0.2 0 0 0</pose>
          <geometry>
            <box>
              <size>0.4 0.1 0.4</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_29">
      <static>true</static>
      <link name="link_29">
        <visual name="visual_29">
          <pose>2.55 3.05 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_29">
          <pose>2.55 3.05 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_30">
      <static>true</static>
      <link name="link_30">
        <visual name="visual_30">
          <pose>0.6 3.15 0.2 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.1 0.4</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_30">
          <pose>0.6 3.15 0.2 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.1 0.4</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_31">
      <static>true</static>
      <link name="link_31">
        <visual name="visual_31">
          <pose>2.65 3.15 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_31">
          <pose>2.65 3.15 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_32">
      <static>true</static>
      <link name="link_32">
        <visual name="visual_32">
          <pose>2.75 3.25 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_32">
          <pose>2.75 3.25 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_33">
      <static>true</static>
      <link name="link_33">
        <visual name="visual_33">
          <pose>2.85 3.35 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_33">
          <pose>2.85 3.35 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_34">
      <static>true</static>
      <link name="link_34">
        <visual name="visual_34">
          <pose>0.5 3.5 0.1 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.2 0.2</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_34">
          <pose>0.5 3.5 0.1 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.2 0.2</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_35">
      <static>true</static>
      <link name="link_35">
        <visual name="visual_35">
          <pose>2.95 3.45 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_35">
          <pose>2.95 3.45 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_36">
      <static>true</static>
      <link name="link_36">
        <visual name="visual_36">
          <pose>3.05 3.55 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_36">
          <pose>3.05 3.55 0.05 0 0 0</pose>
          <geometry>
            <box>
              <size>0.5 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_37">
      <static>true</static>
      <link name="link_37">
        <visual name="visual_37">
          <pose>0.55 3.65 0.1 0 0 0</pose>
          <geometry>
            <box>
              <size>0.1 0.1 0.2</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_37">
          <pose>0.55 3.65 0.1 0 0 0</pose>
          <geometry>
            <box>
              <size>0.1 0.1 0.2</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_38">
      <static>true</static>
      <link name="link_38">
        <visual name="visual_38">
          <pose>0.95 3.65 0.3 0 0 0</pose>
          <geometry>
            <box>
              <size>0.1 0.1 0.6</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_38">
          <pose>0.95 3.65 0.3 0 0 0</pose>
          <geometry>
            <box>
              <size>0.1 0.1 0.6</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_39">
      <static>true</static>
      <link name="link_39">
        <visual name="visual_39">
          <pose>0.2 3.75 0.25 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.1 0.5</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_39">
          <pose>0.2 3.75 0.25 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.1 0.5</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_40">
      <static>true</static>
      <link name="link_40">
        <visual name="visual_40">
          <pose>0.9 3.75 0.3 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.1 0.6</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_40">
          <pose>0.9 3.75 0.3 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.1 0.6</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_41">
      <static>true</static>
      <link name="link_41">
        <visual name="visual_41">
          <pose>0.7 3.85 0.3 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.1 0.6</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_41">
          <pose>0.7 3.85 0.3 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.1 0.6</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_42">
      <static>true</static>
      <link name="link_42">
        <visual name="visual_42">
          <pose>1 3.85 0.3 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.1 0.6</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_42">
          <pose>1 3.85 0.3 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.1 0.6</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

    <model name="box_43">
      <static>true</static>
      <link name="link_43">
        <visual name="visual_43">
          <pose>0.8 3.95 0.3 0 0 0</pose>
          <geometry>
            <box>
              <size>0.6 0.1 0.6</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_43">
          <pose>0.8 3.95 0.3 0 0 0</pose>
          <geometry>
            <box>
              <size>0.6 0.1 0.6</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>

  </world>
</sdf>
